### PR TITLE
Fix hardlink extraction with strip.

### PR DESCRIPTION
**Pull request by bruce-one**
Ensure the code updates linkpath if the hardlink target will have its path
stripped.

Without this fix then the hardlink isn't created when this scenario (a hardlink to a stripped path that is no longer accurate) is encountered.

### CODE REVIEW

**Comment by bruce-one** [comment]
(The tar was updated from:

    hardlink-1
    hardlink-2
    symlink

to

    dir/
    dir/hardlink-1
    dir/hardlink-2
    dir/symlink
    hardlink-1
    hardlink-2
    symlink

to make the test simple to write.)

**Comment by bruce-one** [comment]
(Sorry for the noise - couldn't figure out how the fixtures worked.)

Still working on getting 100% test coverage.

File: lib/unpack.js [hunk 1]
```diff
@@ -178,6 +178,13 @@ class Unpack extends Parser {
       if (parts.length < this.strip)
         return false
       entry.path = parts.slice(this.strip).join('/')
+
+      if (entry.type === 'Link') {
+        const linkparts = entry.linkpath.split(/\/|\\/)
+        /* istanbul ignore else */
```
Comment (isaacs) [inline lib/unpack.js]: Why is istanbul ignoring this `else`?  Should have a test to cover both cases.

File: lib/unpack.js [hunk 2]
```diff
@@ -178,6 +178,13 @@ class Unpack extends Parser {
...
+        if (!(linkparts.length < this.strip))
```
Comment (isaacs) [inline lib/unpack.js]: There is a dedicated comparator to mean "not less-than".  Please use `if (linkparts.length >= this.strip)`.

File: test/unpack.js [hunk 3]
```diff
@@ -171,6 +171,12 @@ t.test('links!', t => {
     unpack.end(data)
     check(t)
   })
+
+  t.test('sync strip', t => {
+    const unpack = new UnpackSync({ cwd: dir, strip: 1 })
+    unpack.end(fs.readFileSync(tars + '/links-strip.tar'))
+    check(t)
+  })
```
Comment (isaacs) [inline test/unpack.js]: Please add async version as well.

**Comment by bruce-one** [comment]
Sorry for all the noise, forgot how the test suite worked.

Happy to rebase (all.....) those commits to clean them up, just let me know :-)

**Comment by isaacs** [comment]
Thanks!  Squashed, landed, and published on v4.4.8.

**Pull request closed without merging**